flake.nix

```diff
@@ -64,13 +67,13 @@
             })
           ];
         }).config.system.build.toplevel;
-      };
+      });
 
       htmlDocs = {
-        nixpkgsManual = jobs.manual;
+        nixpkgsManual = builtins.mapAttrs (_: jobSet: jobSet.manual) jobs;
         nixosManual = (import ./nixos/release-small.nix {
           nixpkgs = self;
-        }).nixos.manual.x86_64-linux;
+        }).nixos.manual;
```

Comment:
Note that before this PR, `nixpkgsManual`/`nixosManual` were derivations (hardcoded to `x86_64-linux`), but now they're an attribute set of derivations, so it would break for any tooling that doesn't expect that. We're getting bitten by Flakes poor system handling here :sweat_smile:

In this particular case though I don't think it's a problem: Flakes doesn't specify any inherent stability promises to output attributes, and these attributes are not even mentioned anywhere in Nixpkgs. If this Flake attribute was consumed by automated tooling for example, we'd have a problem.

So, no change necessary here imo, but just wanted to point it out, because it's very easy to break consumers with this kind of thing.

Reply:
Yeah, I'm not sure why we weren't already having these attributes be attribute sets of derivations before. My guess is most people use `x86_64-linux` so why bother for others systems. I mainly wanted to fix the issue I had with this causing failed jobs on Garnix with forks. But now this opens the door to using these things outside of `x86_64-linux` and not rely on binfmt. Really is the best way to go even if compatibility has to be broken. More people are going to transition to non-x86 systems (including myself) in the future so having proper support is key.

Reply:
Ohh, we can do this to keep `nix build .#htmlDocs.nixpkgsManual` working:

```nix
      packages = forAllSystems (system: {
        htmlDocs = {
          nixpkgsManual = jobs.${system}.manual;
          nixosManual = (import ./nixos/release-small.nix {
            nixpkgs = self;
          }).nixos.manual.${system};
        };
      });
```

Reply:
Does `packages.${system}` even accept multi-level attribute sets? I haven't seen anyone done it so I am not sure.

Reply:
Just tried it and nope, I get an error (`error: expected a derivation`) so I don't think making `htmlDocs` inside of `packages` would be a good solution unless we remove the `htmlDocs` attribute set and just make it `packages.${system}.nix{pkgs,os}Manual`.

Reply:
Oh damn, sorry I apparently wasn't very careful when testing :sweat_smile: